client/scala/src/main/scala/ai/verta/repository/Commit.scala

```diff
@@ -338,4 +338,59 @@ class Commit(
     else
       createCommit(message = message, diffs = diff.blobDiffs, commitBase = id)
   }
+
+  /** Generates a stream folder names and blob names in this commit by walking through its folder tree.
+   *  The stream ends at the first failure, or when there are no folders left. If the commit is not saved, its only element is that Failure
+   *  @return a stream of Try's of WalkOutput.
+   */
+  def walk()(implicit ec: ExecutionContext) = {
+    if (!saved)
+      Stream(Failure(new IllegalCommitSavedStateException("Commit must be saved before it can be walked")))
+    else
+      continueWalk(List(List()))
+  }
+
+  /** Continue the walk from the locations passed
+   *  @param locations remaining locations to explore. Each location is in list of string format.
+   *  @return Stream of Trys of WalkOutputs. If the returned WalkOutput fails, abort the remaining locations.
+   */
+  def continueWalk(locations: List[List[String]])(implicit ec: ExecutionContext): Stream[Try[WalkOutput]] = {
```

Comment:
Use stronger types. I think `List[String]` means something specific here, not a generic list. It also simplifies understanding.

Reply:
Fixed.